presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/EnvironmentUp.java

```diff
@@ -139,12 +142,15 @@ private void killContainersReaperContainer()
             }
         }
 
-        private void sleepUntilInterrupted()
+        private void sleepUntilInterruptedOrContainersStopped(Collection<DockerContainer> containers)
```

Comment:
Maybe this method should accept `Predicate`? Then I would name it `waitUntil`. 
By default any blocking method should handle interruption and it is possibly to throw `InterruptedException`